### PR TITLE
Compile tests added for generation tests

### PR DESCRIPTION
**Pull request by msadikkoseaupa**

### CODE REVIEW

File: src/Nox.Generator/Application/Commands/CreateCommandGenerator.cs [hunk 1]
```diff
@@ -25,10 +25,14 @@ public void Generate(SourceProductionContext context, NoxSolutionCodeGeneratorSt
 
             var primaryKeysQuery = string.Join(", ", entity.Keys.Select(k => $"entityToCreate.{k.Name}.Value"));
 
+            //TestEntityTwoRelationshipsOneToOne has more than one relationship to SecondTestEntityTwoRelationshipsOneToOne
```
Comment (rochar) [inline src/Nox.Generator/Application/Commands/CreateCommandGenerator.cs]: we do not want this comments on the template?

File: src/Nox.Generator/Application/Commands/CreateCommand.template.cs [hunk 2]
```diff
@@ -1,4 +1,3 @@
-﻿{{- relatedEntities = entity.Relationships | array.map "Entity" }}
```
Comment (rochar) [inline src/Nox.Generator/Application/Commands/CreateCommand.template.cs:1]: so were do we compute the related entities?

File: src/Nox.Generator/Application/Commands/CreateCommand.template.cs
Comment (rochar) [inline src/Nox.Generator/Application/Commands/CreateCommand.template.cs:1]: add also the new generated code, so we can see the differences

File: tests/Nox.Integration.Tests/CompileTests.cs
Comment (rochar) [inline tests/Nox.Integration.Tests/CompileTests.cs:1]: Can we move this test to Nox.Generator.Tests?

**Review by rochar — CHANGES_REQUESTED** [review]

File: tests/Nox.Integration.Tests/CompileTests.cs
Comment (msadikkoseaupa) [reply tests/Nox.Integration.Tests/CompileTests.cs:1]: If we copy exactly same nox.yaml files with this project it is possible. Because this project covers many edge cases.

File: src/Nox.Generator/Application/Commands/CreateCommandGenerator.cs (hunk 1 above)
Comment (msadikkoseaupa) [reply src/Nox.Generator/Application/Commands/CreateCommandGenerator.cs]: This a comment for your attention actually, we calculate the related entities here with group by method.

File: src/Nox.Generator/Application/Commands/CreateCommand.template.cs (hunk 2 above)
Comment (msadikkoseaupa) [reply src/Nox.Generator/Application/Commands/CreateCommand.template.cs:1]: we compute related entites in CreateCommandGenerator.cs lines between 28 to 33

**Review by rochar — DISMISSED** [review]

**Review by rochar — APPROVED** [review]

**Review by dejobratic — APPROVED** [review]
